src/Common/ThreadPool.cpp

```diff
@@ -211,16 +215,26 @@ ReturnType ThreadPoolImpl<Thread>::scheduleImpl(Job job, Priority priority, std:
                 /// Most likely this is a std::bad_alloc exception
                 return on_error("cannot allocate thread slot");
             }
+            thread_it = threads.begin();
+
+            ++scheduled_jobs;
+            scheduled_jobs_incremented = true;
+            lock.unlock();
+
 
             try
             {
-                threads.front() = Thread([this, it = threads.begin()] { worker(it); });
+                *thread_it = Thread([this, thread_it] { worker(thread_it); });
```

Comment:
now i remember why I needed that interator holder - if you pass the iterator out of the lock (used to sync access to the `threads` list), the iterator can become incorrect / invalid if the `threads` will be modified by some different thread (for example somebody else will do clear).